core/shared/src/main/scala/fs2/async/mutable/Queue.scala

```diff
@@ -40,12 +40,22 @@ trait Queue[F[_], A] { self =>
   /** Dequeue one `A` from this queue. Completes once one is ready. */
   def dequeue1: F[A]
 
+  /** Dequeue all available `A`s from this queue. Completes once any values are ready. */
+  def dequeueAvailable1: F[Vector[A]]
```

Comment:
The API is by far most useful if it exposes the chunking.  That's sort of the whole point.  :-)  I would be in favor of using `Chunk` rather than `Vector` though.